aip/0180.md

```diff
@@ -0,0 +1,126 @@
+---
+aip:
+  id: 180
+  state: reviewing
+  created: 2019-07-23
+permalink: /180
+redirect_from:
+  - /0180
+---
+
+# Backwards compatibility
+
+APIs are fundamentally contracts with users, and users often write code against
+APIs that is then launched into a production service with the expectation that
+it continues to work (unless the API has a [stability level][aip-181] that
+indicates otherwise). Therefore, it is important to understand what constitutes
+a backwards compatible change and what constitutes a backwards incompatible
+change.
+
+## Guidance
+
+Existing client code **must not** be broken by a service updating to a new
+minor or patch release. Old clients **must** be able to work against newer
+servers (with the same major version number).
+
+**Important:** It is not always salient whether a change is compatible or not.
+The guidance here **should** be treated as indicative, rather than as a
+comprehensive list of every possible change.
+
+There are three distinct types of compatibility to consider:
+
+1. Source compatibility: Code written against a previous version **must**
+   compile against a newer version, and successfully run with a newer version
+   of the client library.
+2. Wire compatibility: Code written against a previous version **must** be able
+   to communicate correctly with a newer server. In other words, not only are
+   inputs and outputs compatible, but the serialization and deserialization
+   expectations continue to match.
+3. Semantic compatibility: Code written against a previous version **must**
+   continue to receive the intended and expected results.
+
```

Comment:
Is it worth mentioning that this sounds significantly more black-and-white than it really is? If the customer has an idea of "expected" results which is "always exactly the same as before" then things like changing the default number of items returned by a list would count as a breaking change. We can maybe do that later. Maybe leave a TODO comment? (If we have a comment option here... I know Markdown isn't great in terms of comments...)

Reply:
This is a great point. Added.
(In this case, HTML comments can be included.)